python/TestHarness/RunParallel.py

```diff
@@ -156,7 +167,20 @@ def returnToTestHarness(self, job_index):
     did_pass = True
 
     output = 'Working Directory: ' + tester.specs['test_dir'] + '\nRunning command: ' + command + '\n'
-    output += self.readOutput(f)
+
+    if tester.specs.isValid('redirect_stdout'):
+      # If the tester enabled redirect_stdout and is using more than one processor
+      if tester.specs['redirect_stdout'] != False \
+         and (self.options.parallel != None and int(self.options.parallel) > 1) \
+         and int(tester.specs['max_parallel']) > 1:
```

Comment:
Don't look at at max_parallel at all. That's is the maximum value for this test. It's not the actual value to use.